consensus/istanbul/backend/backend.go

```diff
@@ -497,9 +498,26 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		return nil
 	}
 
-	if sb.broadcaster != nil {
-		sb.broadcaster.Enqueue(fetcherID, block)
+	// If no cached result, we import block through fetcher, which requires processing the block one more time.
+	// Else if cache hit, we directly write the block.
+	if result == nil {
```

Comment:
What happens if both `result` and `broadcaster` are `nil`? Is that a case that could occur or is it not supposed to occur at all?

Reply:
Good catch!

IMO, it should fail, today it seems to panic due to trying to use the `nil` result

Reply:
Great catch! That is a case that could occur.
I fixed it.

Reply:
Hey @YoduYodu, Is it even possible for result to be nil here? I've looked through the code a bit and it seems to me that to be calling Commit a node must have either proposed the block or verified it and in both of those cases they would have processed the block, meaning that It shouldn't be possible to reach this point with a nil result.

Reply:
@piersy In an edge case where the node goes offline and then back online, we could get a nil result when calling `Commit`

Reply:
@YoduYodu Is that because we are not saving the state in `core.rsSaveDecorator`?  And if so, is it possible we could save the state in  `core.rsSaveDecorator`?

Reply:
That is correct @piersy. This state is in memory, thus it's loss when restarting. We could try to persist it to disk. But i wouldn't recommend that approach. The `stateDB` is not really a `value object`, more of a `transaction object`, so not really designed for persistence. Trying to do that would be quite hard and probably a good source of future bugs. 

And all of this, just to catch an edge case and avoid re-processing a block at that time. So, IMO it's not worth it.

Reply:
Hey @mcortesi I hadn't realised that the `StateProcessResult` has a `StateDB` as one of its fields. I agree it's not worth it here.

My concern about re-processing the block is not that it might happen as an edge case but rather that, a future erroneous code change could cause the result to always be nil and we wouldn't be alerted to that unless we looked at the logs.